repo/blob/storage.go

```diff
@@ -21,6 +21,9 @@ var ErrSetTimeUnsupported = errors.Errorf("SetTime is not supported")
 // ErrInvalidRange is returned when the requested blob offset or length is invalid.
 var ErrInvalidRange = errors.Errorf("invalid blob offset or length")
 
+// ErrBlobImmutableDueToPolicy is returned when a policy prevented an action on an Azure blob.
+var ErrBlobImmutableDueToPolicy = errors.Errorf("blob immutable due to policy")
```

Comment:
i don't think we need to add another error here unless it's expected that callers will need to deal with it somehow.

Reply:
It's just to make `DeleteBlob` cleaner which uses it. Otherwise we'd need to repeat the content of `translateError` inside `DeleteBlob` again 

e.g. inside `DeleteBlob`, instead of just `err = translateError(err)` and the switch with a new case it'd need to be something like:

```
err = translateError(err)

if errors.Is(err, blob.ErrBlobNotFound) {
    // don't return error if blob is already deleted
    return nil
}

var re *azcore.ResponseError

if errors.As(err, &re) && re.ErrorCode == bloberror.BlobImmutableDueToPolicy {
    return az.retryDeleteBlob(ctx, b)
}

```

Reply:
updated

Reply:
At this point this error seems to be specific to Azure.